core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -96,13 +101,18 @@ abstract class DelayedOperation(override val delayMs: Long) extends TimerTask wi
   def tryComplete(): Boolean
 
   /**
-   * Thread-safe variant of tryComplete(). This can be overridden if the operation provides its
-   * own synchronization.
+   * Thread-safe variant of tryComplete() that attempts completion only if the lock can be acquired
+   * without blocking.
    */
   def safeTryComplete(): Boolean = {
```

Comment:
If we don't need the overrides anymore, can this be private? Also, how about `maybeTryComplete`?

Reply:
My mistake. It does need to be exposed.

Reply:
Made it private in `server` and renamed to `maybeTryComplete`.